mono/mini/method-to-ir.c

```diff
@@ -1719,15 +1726,32 @@ mini_emit_memcpy (MonoCompile *cfg, int destreg, int doffset, int srcreg, int so
 MonoInst*
 mono_create_tls_get (MonoCompile *cfg, MonoTlsKey key)
 {
-	gpointer getter = mono_tls_get_tls_getter (key, FALSE);
-	return mono_emit_jit_icall (cfg, getter, NULL);
+	if (cfg->compile_aot) {
+		MonoInst *addr;
+		/*
+		 * tls getters are critical pieces of code and we don't want to resolve them
+		 * through the standard plt/tramp mechanism since we might expose ourselves
+		 * to crashes and infinite recursions.
+		 */
+		EMIT_NEW_AOTCONST (cfg, addr, MONO_PATCH_INFO_GET_TLS_TRAMP, (void*)key);
```

Comment:
@vargaz could you please review this commit ? While testing this commit, I can confirm that the preinit stuff wouldn't work anyway since the plt ignores the preinited got entries and still uses the ones with specific trampolines. Also please let me know if there is anything else to say about this PR.

I plan to squash this commit into the tls refactoring one (to avoid bugs in the intermediary history) and proceed with the merge.